src/components/ProductList.jsx

```diff
@@ -1,11 +1,27 @@
+import { useFilter } from 'contexts/';
+import { filterByGenres, filterByPrice, filterByRatings, filterByBookTypes, sortBooks } from 'utils/';
 import { ProductItem } from './ProductItem';
 
 const ProductList = ({ products }) => {
+
+    const { filterState : { genres, price, ratingsMoreThan, bookType, sortBy } } = useFilter();
+
+    const booksFilteredByGenre = filterByGenres(products, genres);
+    const booksFilteredByPrice = filterByPrice(booksFilteredByGenre, price);
+    const booksFilteredByRatings = filterByRatings(booksFilteredByPrice, ratingsMoreThan);
+    const booksFilteredByBookTypes = filterByBookTypes(booksFilteredByRatings, bookType);
+
+    const booksSortedByOptions = sortBooks(booksFilteredByBookTypes, sortBy);
```

Comment:
you can change this into a single compose function. Its actually doing the same thing as a compose function

Reply:
Okay, I shall keep this in mind for the future.